spec/requests/api/v0/articles_spec.rb

```diff
@@ -342,6 +342,71 @@
     end
   end
 
+  describe "GET /api/articles/:username/:slug" do
+    it "returns CORS headers" do
+      origin = "http://example.com"
+      get slug_api_articles_path(article.username, article.slug), headers: { "origin": origin }
+      expect(response).to have_http_status(:ok)
+      expect(response.headers["Access-Control-Allow-Origin"]).to eq(origin)
+      expect(response.headers["Access-Control-Allow-Methods"]).to eq("HEAD, GET, OPTIONS")
+      expect(response.headers["Access-Control-Expose-Headers"]).to be_empty
+      expect(response.headers["Access-Control-Max-Age"]).to eq(2.hours.to_i.to_s)
+    end
+
+    it "returns correct tag list" do
+      get slug_api_articles_path(username: article.username, slug: article.slug)
+
+      expect(response.parsed_body["tag_list"]).to be_a_kind_of String
```

Comment:
It would probably a good idea to check that the returned tag list actually matches the article tag list rather than just checking if its a string. I also think to save some spec runtime you could merge the spec below into this one and just have a single spec that checks the tags. 

```ruby
    it "returns correct tags" do
      get slug_api_articles_path(username: article.username, slug: article.slug)

      expect(response.parsed_body["tags"]).to match_array(article.tags)
      expect(response.parsed_body["tag_list"]).to eq(article.tag_list) # or whatever method you need to call to get the correct match
    end
```